webui/react/src/pages/TrialDetails/TrialInfoBox.tsx

```diff
@@ -2,26 +2,49 @@ import Card from 'hew/Card';
 import { useModal } from 'hew/Modal';
 import { Loadable, Loaded, NotLoaded } from 'hew/utils/loadable';
 import { isEqual } from 'lodash';
+import { useObservable } from 'micro-observables';
 import React, { useCallback, useEffect, useMemo, useState } from 'react';
 
 import CheckpointModalComponent from 'components/CheckpointModal';
 import ModelCreateModal from 'components/ModelCreateModal';
 import OverviewStats from 'components/OverviewStats';
 import RegisterCheckpointModal from 'components/RegisterCheckpointModal';
+import ResourceAllocationModalComponent from 'components/ResourceAllocationModal';
 import Section from 'components/Section';
 import TimeAgo from 'components/TimeAgo';
-import { getModels } from 'services/api';
+import { SimplifiedNode } from 'pages/ResourcePool/NodeElement';
+import { getModels, getTaskAllocation } from 'services/api';
 import { V1GetModelsRequestSortBy } from 'services/api-ts-sdk';
-import { CheckpointWorkloadExtended, ExperimentBase, ModelItem, TrialDetails } from 'types';
+import clusterStore from 'stores/cluster';
+import {
+  AllocationData,
+  CheckpointWorkloadExtended,
+  ExperimentBase,
+  ModelItem,
+  Resource,
+  ResourcePool,
+  RunState,
+  TrialDetails,
+} from 'types';
 import handleError, { ErrorType } from 'utils/error';
 import { validateDetApiEnum } from 'utils/service';
-import { humanReadableBytes } from 'utils/string';
+import { humanReadableBytes, pluralizer } from 'utils/string';
+
+import css from './TrialInfoBox.module.scss';
 
 interface Props {
   experiment: ExperimentBase;
   trial?: TrialDetails;
 }
 
+interface RpData extends ResourcePool {
+  isRPFull: boolean;
+  isRunning: boolean;
+  name: string;
+  nodes: Array<{ nodeName: string; slotsIds: Resource[] }>;
```

Comment:
why not use `Node` interface you defined in modal? we should define the type in a proper file